sympy/vector/coordsysrect.py

```diff
@@ -173,6 +176,57 @@ def __str__(self, printer=None):
     def __iter__(self):
         return iter([self.i, self.j, self.k])
 
+    def set_coordinate_type(self,  curv_coord_type):
```

Comment:
What about renaming it `connect_to_standard_cartesian(` ?

Also, for the `curv_cood_type`, I would allow to support three possible parameters:

- string
- equation variables and expressions, like `a.set_coordinate_type((r, theta, phi), (r*sin(phi)*cos(theta), ... , ...))`
- lambdas or functions, like `a.set_coordinate_type(lambda r, theta, phi: (r*sin(phi)*cos(theta), ..., ..., ))`